File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int n = nums.size();
        set<int>set;
        for(auto el: nums) set.insert(el);
        int maxLen = 0;
        for(int i=0; i<n; i++){
            if(set.find(nums[i] + 1) != set.end()) continue;
            int len = 0;
            int num = nums[i];
            while(set.find(num) != set.end()){
                num--;
                len++;
            }
            maxLen = max(len, maxLen);
        }
        return maxLen;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set()

        for x in nums:
            s.add(x)

        mx = 0
        for x in nums:
            curr_len = 0
            curr = x
            if x + 1 not in s:
                while curr in s:
                    curr -= 1
                    curr_len += 1
                    mx = max(curr_len, mx)

        return mx
